docs/application/native/guides/machine-learning/machine-learning-inference.md

```diff
@@ -378,6 +431,69 @@ For more information about the pipeline states, see [GStreamer guide](https://gs
     ml_tensors_data_set_tensor_data (data, 0, dummy, 1);
     ```
 
+## Custom Filter
+
+For developers' convenience, NNStreamer provides an interface for processing the tensors data with `custom-easy` framework. After registering the user-defined callback function with input/output tensors information, it can manipulate tensors data in the pipeline without an independent shared object. Since callback function works as _filter_ in the pipeline, it is named as `Custom Filter`.
+
+- **Define and register Custom Filter**
+
+    Before using the Custom Filter in the pipeline, you have to register it with input/output tensors information and its name.
+
+    ```c
+    /* Define Custom Filter function */
+    static int custom_filter_invoke_cb (const ml_tensors_data_h in, ml_tensors_data_h out, void *user_data)
+    {
+        /* Get input tensors using data handle 'in',
+           and fill output tensors using data handle 'out'. */
+        if (user_data) {
+            void *raw_data = NULL;
+            size_t *data_size = (size_t *) user_data;
+
+            ml_tensors_data_get_tensor_data (out, 0, &raw_data, data_size);
+        }
+
+        return 0;
+    }
+    ...
+
+    ml_tensors_info_h in_info, out_info;
+    ml_custom_easy_filter_h custom;
+
+    /* Define input and output tensors information. */
+    ml_tensors_info_create (&in_info);
+    ml_tensors_info_set_count (in_info, 1);
+    ml_tensors_info_set_tensor_type (in_info, 0, ML_TENSOR_TYPE_INT8);
+    ml_tensors_info_set_tensor_dimension (in_info, 0, dim);
+
+    ml_tensors_info_create (&out_info);
+    ml_tensors_info_set_count (out_info, 1);
+    ml_tensors_info_set_tensor_type (out_info, 0, ML_TENSOR_TYPE_FLOAT32);
+    ml_tensors_info_set_tensor_dimension (out_info, 0, dim);
+
+    /* Register custom filter with name 'my-custom-filter' */
+    status = ml_pipeline_custom_easy_filter_register ("my-custom-filter", in_info, out_info, custom_filter_invoke_cb, NULL, &custom);
```

Comment:
Instead of NULL, you need to provide ```((void *) &data_size)```, where ```size_t data_size = some value or get the size with ml-tensors_info_get_tensor_size()```

Reply:
ACK!. My falut!